src/NosnocSolver.m

```diff
@@ -431,18 +444,44 @@ function  print_solver_stats(obj, results, stats)
 
                 % Using multi-solver
                 if iscell(solver)
-                    start = cputime
+                    start = cputime;
+                    tic;
                     nlp_results = solver{ii+1}('x0', w0,...
                         'lbx', lbw,...
                         'ubx', ubw,...
                         'lbg', lbg,...
                         'ubg', ubg,...
                         'p',obj.p_val);
                     cpu_time_iter = cputime - start;
+                    wall_time_iter = toc;
                     stats.solver_stats = [stats.solver_stats, solver{ii+1}.stats];
                 else
                     if ii ~= 0
+
+                        % TODO(Anton) Lets push on casadi devs to allow for changing of options after construction
+                        %             (or maybe do it ourselves) and then remove this hack
+                        if obj.solver_options.timeout_cpu
+                            casadi_nlp = struct('f', obj.nlp.cost, 'x', obj.nlp.w, 'g', obj.nlp.g, 'p', obj.nlp.p);
+                            if strcmp(solver_options.solver, 'ipopt')
+                                opts_casadi_nlp = rmfield(solver_options.opts_casadi_nlp, 'snopt');
+                                opts_casadi_nlp.ipopt.max_cpu_time = solver_options.timeout_cpu - stats.cpu_time_total;
+                            elseif strcmp(solver_options.solver, 'snopt')
+                                opts_casadi_nlp = rmfield(solver_options.opts_casadi_nlp, 'ipopt');
+                            end
+                            solver = nlpsol(solver_options.solver_name, solver_options.solver, casadi_nlp, opts_casadi_nlp);
+                        elseif obj.solver_options.timeout_wall
+                            casadi_nlp = struct('f', obj.nlp.cost, 'x', obj.nlp.w, 'g', obj.nlp.g, 'p', obj.nlp.p);
+                            if strcmp(solver_options.solver, 'ipopt')
+                                opts_casadi_nlp = rmfield(solver_options.opts_casadi_nlp, 'snopt');
```

Comment:
This line is removing the snopt field from the casadi nlp options. This is because casadi will freak out if you pass an options struct that contains unexpected fields.